src/nvim/main.c

```diff
@@ -1023,10 +1023,6 @@ static void command_line_scan(mparm_T *parmp)
           want_argument = true;
           break;
         }
-        case 'Z': {  // "-Z" restricted mode
```

Comment:
Do we want a dedicated error message here (like a reference to `|nvim-features-removed|`) or is the automatic `nvim: Unknown option argument: "-Z"` enough? cc @justinmk